File: cliente_rest.php
<?php
declare(strict_types = 1);
$baseUrl=   "https://api.sifei.com.mx/cfdipdf/api/v2/";

#se prepara la URL relativa para el metodo(endpoint)
$urlMethod=$baseUrl."pdf";
echo "consumiendo(consuming):".$urlMethod."\n";
$token="TU_TOKEN";
#se preparan los datos como color, xml y logo.
$params =  [
    'primaryColor'=>'#1599BF',
    'secundaryColor'=>'#E7A201',
    'xml'=>  base64_encode(file_get_contents( __DIR__."/IngresoV2_DEMO.xml")),
    'logo'=> base64_encode(file_get_contents( __DIR__.'/logo.jpg')), 
];
$ch = curl_init();
curl_setopt_array($ch, 
array(
CURLOPT_URL            => $urlMethod,
CURLOPT_RETURNTRANSFER => true,
CURLOPT_POST           => 1,
CURLOPT_HTTPHEADER     => array(
                            'Content-Type:application/json',
                            'Authorization:'.$token, #token                            
                        ),
CURLOPT_POSTFIELDS     => json_encode($params)
));
$resultCurl = curl_exec($ch);
$httpcode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
curl_close($ch);

if ($httpcode!=200) {
    if (!$resultCurl) {
        #Error de red, de resolucion de dominio,etc
        echo "Un error no controlado ocurrio";
        return;
    }
}else{
    echo "Codigo exitoso, recuperado PDF generado.(Successfull code, retrieving generated pdf from response.";
    $arrRes=json_decode($resultCurl, true);  

    #se escribe a archivo el PDF contenido en el atributo data de la respuesta.
    file_put_contents(__DIR__."/".$arrRes['uuid']. time().'.pdf', base64_decode($arrRes['data']));
    echo "creado";    
}

File: cliente_rest.py
import http.client
import json
import base64

server="api.sifei.com.mx"


conn = http.client.HTTPSConnection(server)

cfdiPath="cliente/XML_DEMO.xml"
logoPath='cliente/logo.jpg'
token='TU_TOKEN'
pdfPath=cfdiPath.replace('.xml','.pdf')
with open(cfdiPath,"rb") as cfdiFile:
    base64CFDI=base64.b64encode(cfdiFile.read())
with open(logoPath,"rb") as logoFile:
    base64Logo=base64.b64encode(logoFile.read())
#Preparamos la solicitud    
payload={
    'primaryColor':'#1599BF',
    'secundaryColor':'#E7A201',
    'logo': str(base64Logo.decode('utf-8')), 
    'xml': str(base64CFDI.decode('utf-8'))
}
headers = {
    'Content-Type': "application/json",
    'Authorization': token
}
print("Connectando")
conn.request("POST", "/cfdipdf/api/v2/pdf", json.dumps(payload), headers)

res = conn.getresponse()
data = res.read()
print(data)
resul=json.loads(data.decode("utf-8"))

print("httpcode:"+str(res.status))
# the code when it generates a pdf is 200.
if res.status != 200:
    print("Un error ocurrio, codigo:[{}], mensaje de error:[{}]".format(str(resul['code']),resul['message']))

    #400 invalid request
    #401. Wron token, invalid token, expired token
    #500 server error.

else:    
    print("PDF creado. Extrayendo respuesta")
    bPDF=base64.b64decode(resul['data'])
    #escribimos el PDF
    with open(pdfPath,'wb') as pdfFile:
        pdfFile.write(bPDF)
    print("PDF escrito en: {}".format(pdfPath))
       

        